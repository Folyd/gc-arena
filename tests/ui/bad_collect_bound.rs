use gc_arena::Collect;

struct NotCollect;

#[derive(Collect)]
#[collect(no_drop)]
struct MyStruct {
    field: NotCollect
}

fn main() {}
