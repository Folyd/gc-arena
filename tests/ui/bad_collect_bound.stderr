error[E0277]: the trait bound `NotCollect: Collect` is not satisfied
 --> tests/ui/bad_collect_bound.rs:8:12
  |
8 |     field: NotCollect
  |            ^^^^^^^^^^
  |            |
  |            the trait `Collect` is not implemented for `NotCollect`
  |            the trait `Collect` is not implemented for `NotCollect`
  |
  = help: the following other types implement trait `Collect`:
            bool
            char
            isize
            i8
            i16
            i32
            i64
            usize
          and $N others

error[E0277]: the trait bound `NotCollect: Collect` is not satisfied
 --> tests/ui/bad_collect_bound.rs:8:5
  |
5 | #[derive(Collect)]
  |          ------- in this derive macro expansion
...
8 |     field: NotCollect
  |     ^^^^^ the trait `Collect` is not implemented for `NotCollect`
  |
  = help: the following other types implement trait `Collect`:
            bool
            char
            isize
            i8
            i16
            i32
            i64
            usize
          and $N others
  = note: this error originates in the derive macro `Collect` (in Nightly builds, run with -Z macro-backtrace for more info)
